wp-remember-the-galleries/wp-remember-the-galleries.php

```diff
@@ -308,50 +313,74 @@ static function save_gallery() {
 			wp_send_json_error( 'empty-name' );
 		}
 
+		$term_id = null; 
+		if ( isset( $_POST['term_id'] ) ) {
+			$term_id = (int) $_POST['term_id'];
+		}
+
 		$gallery_name = sanitize_text_field( trim( $_POST['name'] ) );
 
 		if( empty( $gallery_name ) ) {
 			wp_send_json_error( 'empty-name' );
 		}
 
-		// Get or insert the appropriate term for this gallery
+		// Does a gallery with this name already exist?
 		$term_info = term_exists( $gallery_name, self::entity );
+		$allow_rename = true;
+		if ( isset( $term_info['term_id'] ) && $term_info['term_id'] != $term_id ) {
+			// Ask user to confirm that they want to overwrite the existing gallery
+			if ( !isset( $_POST['yes'] ) ) {
+				wp_send_json_error( 'need-confirm' );
+			}
+			// User has confirmed they want to overwrite the existing gallery, so let's use that term id
+			else {
+				$allow_rename = false;
+				$term_id = $term_info['term_id'];
+			}
+		}
 
-		if( !isset( $term_info['term_id'] ) ) {
-			$term_info = wp_insert_term( $gallery_name, self::entity, array( 'slug' => self::entity . '--' ) );
+		if( !$term_id ) {
+			$term_info = wp_insert_term( $gallery_name, self::entity, array( 'slug' => self::entity . '-' . $post_id ) );
```

Comment:
$post_id isn't known at this point – This is the first reference to that variable in this scope. I had the placeholder term with '--' suffix on purpose because of this, which is meant to be updated later in the code after the post is inserted and $post_id is known. I bet you'll get a PHP Warning when this code path hits.